src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1358,150 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromClassnames(
+      Set<@ClassGetName String> classnames, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> dependenciesClassnames = new TreeSet<>();
+
+    for (String classname : classnames) {
+      try {
+        Class<?> getDependenciesFrom = Class.forName(classname);
+        for (Method method : getDependenciesFrom.getDeclaredMethods()) {
+          addMethodParameterTypesIfShould(method, dependenciesClassnames, accessibility);
+        }
+        for (Constructor<?> constructor : getDependenciesFrom.getConstructors()) {
+          addConstructorParameterTypesIfShould(constructor, dependenciesClassnames, accessibility);
+        }
+      } catch (ClassNotFoundException e) {
+        throw new RandoopUsageError(String.format("Cannot load class %s", classname));
+      }
+    }
+    return dependenciesClassnames;
+  }
+
+  /**
+   * Returns names of classes methods in methodlist depend on. Does not add dependencies of methods
+   * that should be omitted. Does not add classes that are not accessible or should be omitted.
+   *
+   * @param accessibilityPredicate an accessibility predicate
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromMethodList(
+      AccessibilityPredicate accessibilityPredicate) {
+    Set<@ClassGetName String> classnames = new TreeSet<>();
+    if (methodlist == null) {
+      return Collections.emptySet();
+    }
+
+    ReflectionPredicate reflectionPredicate = new DefaultReflectionPredicate();
+    try (EntryReader reader = new EntryReader(methodlist, "(//|#).*$", null)) {
+      for (String signature : reader) {
+        if (!shouldOmitMethod(signature)) {
+          AccessibleObject accessibleObject = null;
+          try {
+            accessibleObject =
+                SignatureParser.parse(signature, accessibilityPredicate, reflectionPredicate);
+          } catch (SignatureParseException | FailedPredicateException e) {
+            continue;
+          }
+          if (accessibleObject instanceof Constructor) {
+            addConstructorParameterTypesIfShould(
+                (Constructor<?>) accessibleObject, classnames, accessibilityPredicate);
+          }
+          if (accessibleObject instanceof Method) {
+            addMethodParameterTypesIfShould(
+                (Method) accessibleObject, classnames, accessibilityPredicate);
+          }
+        }
+      }
+    } catch (IOException e) {
+      throw new RandoopUsageError("Can`t read methods from " + methodlist.toString());
+    }
+    return classnames;
+  }
+
+  /**
+   * Tests whether a method should be omitted. Returns true if the signature matches any pattern in
+   * omit_method list, false otherwise.
+   *
+   * @param signature signature of method to test
+   * @return true if method should be ommited
+   */
+  private static boolean shouldOmitMethod(String signature) {
```

Comment:
This implementation is nice and simple, but it does not respect the Randoop documentation, which requires checking other classes in the type hierarchy: https://randoop.github.io/randoop/manual/#option:omit-methods .

I think it is better to make a regular `OmitMethodsPredicate` rather than reimplementing part of its functionality.